mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -295,7 +296,8 @@ private void allowOnMailboxSession(List<MailboxId> mailboxIds, MailboxSession ma
             .findFirst();
 
         if (mailboxForbidden.isPresent()) {
-            throw new MailboxNotFoundException("Mailbox with Id " + mailboxForbidden.get() + " does not belong to session");
+            LOGGER.info("Mailbox with Id " + mailboxForbidden.get() + " does not belong to " + mailboxSession.getUser().getUserName());
+            throw new MailboxNotFoundException(mailboxForbidden.get());
```

Comment:
no integration test to see the change? Anyway I don't see how we don't disclose anymore in this commit.

Reply:
The JMAP layer relies on the exception message for computing the JMAP error sent to the client.

If the message contains: "does not belong to alice"

Then alice knows the folder exist :-)

If you want me to add integration test for this...

Reply:
ok so you are still returning the mailboxId, but as you don't tell that it's forbidden because it's beloging to an other user, you get the same error message than for a non existent mailbox.
Then fine for me.

Reply:
Yes that's the idea.

Reply:
And the LOG is here to not lose track of it.

Reply:
I think we should add an integration test to check we do not disclose yes

Reply:
Crap... A not yet implemented exception is thrown before the MailboxNotFound exception. (Create a message somewhere else than in the outbox)

I corrected some never used code... I'd propose to keep it like this...